util/db_info_dumper.cc

```diff
@@ -118,7 +118,7 @@ void DumpDBFileSummary(const DBOptions& options, const std::string& dbname) {
           char str[8];
           snprintf(str, sizeof(str), "%" PRIu64, file_size);
           wal_info.append(file).append(" size: ").
-              append(str, sizeof(str)).append(" ;");
+              append(str).append(" ; ");
```

Comment:
Hmm, appending a string and expecting it's going to be null-terminated is a bit risky. How about something like this:

     wal_info = wal_info + file + " size: " + std::to_string(file_size) + " ; ";